prog.go

```diff
@@ -135,7 +137,11 @@ func NewProgramWithOptions(spec *ProgramSpec, opts ProgramOptions) (*Program, er
 	handles := newHandleCache()
 	defer handles.close()
 
-	return newProgramWithOptions(spec, opts, handles)
+	prog, err := newProgramWithOptions(spec, opts, handles)
+	if errors.Is(err, errUnsatisfiedReference) {
+		return nil, fmt.Errorf("possible attempt to load program instead of collection: %w", err)
```

Comment:
This will come out as `possible attempt to load program instead of collection: unsatisfied reference`. I think we could drop the word 'possible' here: `cannot load program without loading its whole collection: unsatisfied reference to map`.

On the off chance the caller tries to load a bare prog with a missing map ref (no collection), there's still the 'unsatisfied reference to map' to help out as the more-specific error.

Reply:
sgtm.